coderd/database/sqlxqueries/sqlxqueries.go

```diff
@@ -0,0 +1,55 @@
+package sqlxqueries
+
+import (
+	"bytes"
+	"embed"
+	"sync"
+	"text/template"
+
+	"golang.org/x/xerrors"
+)
+
+//go:embed *.gosql
+var sqlxQueries embed.FS
+
+var (
+	// Only parse the queries once.
+	once   sync.Once
+	cached *template.Template
+	//nolint:errname
+	cachedError error
+)
+
+// loadQueries parses the embedded queries and returns the template.
+// Results are cached.
+func loadQueries() (*template.Template, error) {
+	once.Do(func() {
+		tpls, err := template.New("").
+			Funcs(template.FuncMap{
+				"int32": func(i int) int32 { return int32(i) },
+			}).ParseFS(sqlxQueries, "*.gosql")
+		if err != nil {
+			cachedError = xerrors.Errorf("developer error parse sqlx queries: %w", err)
+			return
+		}
+		cached = tpls
+	})
+
+	return cached, cachedError
```

Comment:
This should panic or cause the app to exit if these fail to load as it's unrecoverable and will cause future queries to fail.

Reply:
I don't like doing `init()` functions because then anyone who imports this will have it called. 

I could have this function call it to panic? If you make a new sql store and the sql queries fail to load, then it panics. How does that work?
https://github.com/coder/coder/blob/c2787e3a8e4d53bd7d762d2b9412ae7db791ad30/coderd/database/db.go#L43-L43

Reply:
`init` only gets called once ever, even if you import it multiple times.

Reply:
Yes I know. Still sucks to run code on import.

I can add it to `New` which wraps our SQL db.

Reply:
Sure

Reply:
Done